physics/rrtmgp_sw_gas_optics.F90

```diff
@@ -19,269 +78,393 @@ module rrtmgp_sw_gas_optics
-         kmajor                             ! Stored absorption coefficients due to major absorbing gases
-    character(len=32),  dimension(:), allocatable :: &
-         gas_names,                       & ! Names of absorbing gases
-         gas_minor,                       & ! Name of absorbing minor gas
-         identifier_minor,                & ! Unique string identifying minor gas
-         minor_gases_lower,               & ! Names of minor absorbing gases in lower atmosphere
-         minor_gases_upper,               & ! Names of minor absorbing gases in upper atmosphere
-         scaling_gas_lower,               & ! Absorption also depends on the concentration of this gas
-         scaling_gas_upper                  ! Absorption also depends on the concentration of this gas
-    logical(wl), dimension(:), allocatable :: &
-         minor_scales_with_density_lower, & ! Density scaling is applied to minor absorption coefficients
-         minor_scales_with_density_upper, & ! Density scaling is applied to minor absorption coefficients
-         scale_by_complement_lower,       & ! Absorption is scaled by concentration of scaling_gas (F) or its complement (T)
-         scale_by_complement_upper          ! Absorption is scaled by concentration of scaling_gas (F) or its complement (T)
-    ! Dimensions
-    integer :: &
-         ntemps, npress, ngpts, nabsorbers, nextrabsorbers,       &
-         nminorabsorbers, nmixingfracs, nlayers, nbnds, npairs,   &
-         nminor_absorber_intervals_lower, nminor_absorber_intervals_upper, &
-         ncontributors_lower, ncontributors_upper
 
     ! Local variables
-    integer :: status, ncid, dimid, varID, iGas
+    integer :: status, ncid, dimid, varID, iGas, mpierr, iChar
     integer,dimension(:),allocatable :: temp1, temp2, temp3, temp4
     character(len=264) :: sw_gas_props_file
 
     ! Initialize
     errmsg = ''
     errflg = 0
 
-    ! Filenames are set in the gphysics_nml
+    ! Filenames are set in the gfphysics_nml
     sw_gas_props_file   = trim(rrtmgp_root_dir)//trim(rrtmgp_sw_file_gas)
 
-    ! Read dimensions for k-distribution fields (only on master processor(0))
-!    if (mpirank .eq. mpiroot) then
+    ! #######################################################################################
+    !
+    ! Read dimensions for k-distribution fields...
+    ! (ONLY master processor(0), if MPI enabled)
+    !
+    ! #######################################################################################
+#ifdef MPI
+    if (mpirank .eq. mpiroot) then
+#endif
+       write (*,*) 'Reading RRTMGP shortwave k-distribution metadata ... '
+
        ! Open file
        status = nf90_open(trim(sw_gas_props_file), NF90_NOWRITE, ncid)
 
        ! Read dimensions for k-distribution fields
        status = nf90_inq_dimid(ncid, 'temperature', dimid)
-       status = nf90_inquire_dimension(ncid, dimid, len=ntemps)
+       status = nf90_inquire_dimension(ncid, dimid, len=ntempsSW)
```

Comment:
The code should check the return status and report failures, not just ignore them.